### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by shraddha-amber**
This PR contains:
1. Login, SignUp, Profile Pages
2. User can login using test credentials
3. Profile page contains the logout CTA
4. Basic validation applied

image
image

### CODE REVIEW

**Review by ragini07 — COMMENTED** [review]
Good naming convention . Liked that you created constants for routes and action type. Clean UI

**Review by ragini07 — APPROVED** [review]

File: src/utils/services/authHandler.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      userInfo:
+        authAction === "LOGIN"
+          ? { ...response.data.foundUser }
+          : { ...response.data.createdUser },
+      token: response.data.encodedToken,
+    };
+
+    localStorage.setItem("data", JSON.stringify(data));
+    if (response.status === 201 || response.status === 200)
+      authDispatch({
```
Comment (JowelTisso) [inline src/utils/services/authHandler.js]: * You can extract the localStorage as a separate function
* Also you can do something like this `response.status === 201 || 200`

**Review by JowelTisso — APPROVED** [review]
* Also good and clean UI
* Rest Looks Good

File: src/utils/services/authHandler.js (hunk 1 above)
Comment (shraddha-amber) [reply src/utils/services/authHandler.js]: yes, thank you jowel, i refactored the code now

Comment (JowelTisso) [reply src/utils/services/authHandler.js]: > yes, thank you jowel, i refactored the code now

Correction `response.status === 201 || 200` this code will not work, instead your previous code will work fine i.e `response.status === 201 || response.status === 200`. Sorry for wrong information.

**Pull request merged**